scaleway/resource_object_bucket.go

```diff
@@ -468,15 +470,13 @@ func resourceScalewayObjectBucketRead(ctx context.Context, d *schema.ResourceDat
 		Bucket: aws.String(bucketName),
 	})
 	if err != nil {
-		switch {
-		case isS3Err(err, ErrCodeObjectLockConfigurationNotFoundError, ""):
-			_ = d.Set("object_lock_enabled", false)
-		case isS3Err(err, s3.ErrCodeNoSuchBucket, ""):
-			tflog.Error(ctx, fmt.Sprintf("Bucket %q was not found - removing from state!", bucketName))
+		bucketFound, objectLockFound := addErrorDiagnostic(&diags, err, "object lock configuration", ErrCodeObjectLockConfigurationNotFoundError)
```

Comment:
the `resource` value here can lead to `cty.GetAttrPath` on an invalid input.
This is the same after, I skip creating multiple comments on this